### PR TITLE
EWPP-4356: Use source key of picture parameter for rendering different image sizes for each ECL breakpoint in Banner and Carousel.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

**Comment by fmmribeiro** [comment]
Can the readme file be updated?
https://github.com/openeuropa/oe_theme?tab=readme-ov-file#image-styles

File: modules/oe_theme_helper/oe_theme_helper.post_update.php [hunk 1]
```diff
@@ -326,3 +326,50 @@ function oe_theme_helper_post_update_20017() {
...
+      'height' => '332',
+    ],
+    'oe_theme_small_banner' => [
+      'label' => 'Small banner',
+      'width' => '480',
+      'height' => '160',
+    ],
+  ];
+  foreach ($image_styles as $style_id => $style_data) {
+    $style = \Drupal::entityTypeManager()->getStorage('image_style')->load($style_id);
```
Comment (nagyad) [inline modules/oe_theme_helper/oe_theme_helper.post_update.php]: Let's instantiate a `$storage = \Drupal::entityTypeManager()->getStorage('image_style');` outside of the foreach.

File: oe_theme.theme [hunk 2]
```diff
@@ -1669,14 +1669,35 @@ function oe_theme_preprocess_paragraph__oe_banner(array &$variables): void {
     $cacheability->applyTo($variables);
     return;
   }
-  $style = \Drupal::entityTypeManager()->getStorage('image_style')->load('oe_theme_full_width');
+  $full_width = \Drupal::entityTypeManager()->getStorage('image_style')->load('oe_theme_full_width');
+  $large = \Drupal::entityTypeManager()->getStorage('image_style')->load('oe_theme_large_banner');
+  $medium = \Drupal::entityTypeManager()->getStorage('image_style')->load('oe_theme_medium_banner');
+  $small = \Drupal::entityTypeManager()->getStorage('image_style')->load('oe_theme_small_banner');
```
Comment (nagyad) [inline oe_theme.theme]: `$storage = \Drupal::entityTypeManager()->getStorage('image_style');`

File: oe_theme.theme [hunk 3]
```diff
@@ -2202,6 +2223,12 @@ function oe_theme_preprocess_paragraph__oe_carousel(array &$variables): void {
   $variables['items'] = [];
   $cacheability = CacheableMetadata::createFromRenderArray($variables);
 
+  // Load needed images styles.
+  $full_width = \Drupal::entityTypeManager()->getStorage('image_style')->load('oe_theme_full_width');
+  $large = \Drupal::entityTypeManager()->getStorage('image_style')->load('oe_theme_large_banner');
+  $medium = \Drupal::entityTypeManager()->getStorage('image_style')->load('oe_theme_medium_banner');
+  $small = \Drupal::entityTypeManager()->getStorage('image_style')->load('oe_theme_small_banner');
```
Comment (nagyad) [inline oe_theme.theme]: `$storage = \Drupal::entityTypeManager()->getStorage('image_style')`

File: tests/src/PatternAssertions/CarouselAssert.php [hunk 4]
```diff
@@ -91,6 +92,18 @@ protected function assertItems(array $expected_items, Crawler $crawler): void {
           $this->assertEquals('', $image_element->attr('alt'));
         }
       }
+      if (!isset($expected_item['sources']) ||
+        (isset($expected_item['variant']) && $expected_item['variant'] === 'plain-background')) {
```
Comment (nagyad) [inline tests/src/PatternAssertions/CarouselAssert.php]: This can go in one line.

**Review by nagyad — CHANGES_REQUESTED** [review]

**Review by fmmribeiro — DISMISSED** [review]

**Review by nagyad — DISMISSED** [review]

**Review by nagyad — DISMISSED** [review]

**Review by fmmribeiro — APPROVED** [review]

**Review by nagyad — APPROVED** [review]

**Pull request merged**